src/main/java/net/jonathangiles/tools/teenyhttpd/annotations/PathVariable.java

```diff
@@ -0,0 +1,17 @@
+package net.jonathangiles.tools.teenyhttpd.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation is used to mark a method parameter as a path variable. The value of the annotation is the name of the
+ * path variable to be extracted from the URL.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.PARAMETER)
+public @interface PathVariable {
```

Comment:
Elsewhere I refer to this as 'path parameters' - I wonder if we should call this `PathParam`?

Reply:
Sure, PathParam is good

Reply:
As I've experienced on TeenyHttpd, things churn a lot over time. I'm sure the same will happen here too! Let's get your stuff as polished as possible, and then we can get it merged and iterate over time.